lib/services/app.go

```diff
@@ -72,31 +72,53 @@ type Applications interface {
 
 // ValidateApp validates the Application resource.
 func ValidateApp(app types.Application, proxyGetter ProxyGetter) error {
+	// If no public address is set, there's nothing to validate.
+	if app.GetPublicAddr() == "" {
+		return nil
+	}
+
+	// It is assumed that the app's public address has already been validated to be a valid address format during app
+	// resource validation. The remainder of this function focuses on detecting conflicts with proxy public addresses.
+	appAddr, err := utils.ParseAddr(app.GetPublicAddr())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Convert the application's public address hostname to its ASCII representation for comparison. Strip any trailing
+	// dot to ensure consistent comparison.
+	asciiAppHostname, err := idna.ToASCII(strings.TrimSuffix(appAddr.Host(), "."))
+	if err != nil {
+		return trace.WrapWithMessage(err, "app %q has an invalid IDN hostname %q", app.GetName(), appAddr.Host())
```

Comment:
The trace API is a bit of a mess, but it's generally preferred to use Wrap over WrapWithMessage due to formatting applied in the latter. 
```suggestion
		return trace.Wrap(err, "app %q has an invalid IDN hostname %q", app.GetName(), appAddr.Host())
```